meta/include/gnuradio-4.0/meta/UncertainValue.hpp

```diff
@@ -0,0 +1,331 @@
+    return lhs;
+}
+
+template<typename T, typename U>
+    requires UncertainValueLike<T> || UncertainValueLike<U>
+[[nodiscard]] inline constexpr auto
+operator/(const T &lhs, const U &rhs) noexcept {
+    using ValueTypeT = UncertainValueType_t<T>;
+    using ValueTypeU = UncertainValueType_t<U>;
+    if constexpr (UncertainValueLike<T> && UncertainValueLike<U>) {
+        using ResultType = decltype(lhs.value * rhs.value);
+        if constexpr (complex_like<ValueTypeT> || complex_like<ValueTypeU>) {
+            // we are dealing with complex numbers -> use standard uncorrelated calculation
+            ResultType newUncertainty;
+            if constexpr (std::is_arithmetic_v<ValueTypeT> && complex_like<ValueTypeU>) {
+                // LHS is real, RHS is complex
+                newUncertainty = { std::sqrt(std::pow(lhs.uncertainty / std::real(rhs.value), 2)), std::sqrt(std::pow(std::imag(rhs.uncertainty) * lhs.value / std::norm(rhs.value), 2)) };
+            } else if constexpr (complex_like<ValueTypeT> && std::is_arithmetic_v<ValueTypeU>) {
+                // LHS is complex, RHS is real
+                newUncertainty = { std::hypot(std::real(lhs.uncertainty) / rhs.value, rhs.uncertainty * std::real(lhs.value) / std::pow(rhs.value, 2)),
+                                   std::sqrt(std::pow(std::imag(lhs.uncertainty) / rhs.value, 2)) };
+            } else {
+                newUncertainty = { std::hypot(std::real(lhs.uncertainty) / std::real(rhs.value), std::real(rhs.uncertainty) * std::real(lhs.value) / std::norm(rhs.value)),
+                                   std::hypot(std::imag(lhs.uncertainty) / std::imag(rhs.value), std::imag(rhs.uncertainty) * std::imag(lhs.value) / std::norm(rhs.value)) };
+            }
+
+            return UncertainValue<ResultType>{ lhs.value / rhs.value, newUncertainty };
+        } else {
+            // both ValueType[T,U] are arithmetic uncertainties
+            auto combinedUncertainty = std::hypot(lhs.uncertainty / rhs.value, rhs.uncertainty * lhs.value / std::pow(rhs.value, 2));
+            return UncertainValue<ResultType>{ lhs.value / rhs.value, combinedUncertainty };
+        }
+    } else if constexpr (UncertainValueLike<T> && arithmetic_or_complex_like<ValueTypeU>) {
+        return T{ lhs.value / rhs, lhs.uncertainty / std::abs(rhs) };
+    } else if constexpr (arithmetic_or_complex_like<ValueTypeT> && UncertainValueLike<U>) {
+        auto rhsMagSquared = std::norm(rhs.value);
+        return U{ lhs / rhs.value, rhs.uncertainty * std::abs(lhs) / rhsMagSquared };
+    } else {
+        static_assert(std::is_arithmetic_v<ValueTypeT> && std::is_arithmetic_v<ValueTypeU>);
+        return lhs / rhs; // unlikely to be called due to default '/' definition
+    }
+}
+
+template<UncertainValueLike T, typename U>
+inline constexpr T &
+operator/=(T &lhs, const U &rhs) noexcept {
+    lhs = lhs / rhs;
+    return lhs;
+}
+
+} // namespace gr::meta
+
+namespace std { // std:: basic math overloads
```

Comment:
Leave it in the the gr::meta namespace please. Defining things in std:: is generally UB, and allowed only for specific situations. I can't find that that pow ect. were meant to be customization points.

Reply:
TODO: check with @mattkretz  whether overloading std math functions with user-defined types is acceptable or not

Example:
```cpp
template<gr::UncertainValueLike T, typename ValueTypeT = gr::UncertainValueType_t<T>>
[[nodiscard]] inline constexpr T
std::sin(const T &x) noexcept {
    return T{ std::sin(x.value), std::abs(std::cos(x.value) * x.uncertainty) };
}
```

Reply:
moved to `namespace gr::math` -- don't want to break the world or annoy the C++ gods.